### PR TITLE
Add doc-viewer component

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/components/doc-viewer/doc-viewer.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  /** Instantiate a MaterialIoExampleViewer for each example. */
+  private _initExamples() {
+    let exampleElements =
+        this._elementRef.nativeElement.querySelectorAll('[material-docs-example]');
+    exampleElements.forEach((element: Element) => {
+      let example = element.getAttribute('material-docs-example');
+      console.log(example);
+      let portalHost = new DomPortalHost(element, this._componentFactoryResolver);
```
Comment (mmalerba) [inline src/components/doc-viewer/doc-viewer.ts]: I think there's something weird going on with npm. In package.json if I specify: 
```
"@angular/material": "^2.0.0-alpha.9-3",
```
or
```
"@angular/material": "^2.0.0-alpha.10",
```
it wants to install alpha 9. I can force it by just setting:
```
"@angular/material": "2.0.0-alpha.10",
```

Comment (jelbourn) [inline src/components/doc-viewer/doc-viewer.ts]: Yeah, `npm` doesn't interpret `10 > 9-3`

File: src/components/doc-viewer/doc-viewer.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  template: 'Loading document...',
+})
+export class MaterialIoDocViewer {
+  constructor(private _componentFactoryResolver: ComponentFactoryResolver,
+              private _elementRef: ElementRef,
+              private _http: Http,
+              private _viewContainerRef: ViewContainerRef) {}
+
+  /** The url of the document to display. */
+  @Input('document-url')
```
Comment (mmalerba) [inline src/components/doc-viewer/doc-viewer.ts]: I've seen this in the material2 repo too, should we switch those to also be camelCase?

File: src/components/doc-viewer/doc-viewer.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            `Failed to load document: ${url}. Error: ${error}`;
+      }
+    );
+  }
+
+  /** Instantiate a MaterialIoExampleViewer for each example. */
+  private _initExamples() {
+    let exampleElements =
+        this._elementRef.nativeElement.querySelectorAll('[material-docs-example]');
+    exampleElements.forEach((element: Element) => {
```
Comment (mmalerba) [inline src/components/doc-viewer/doc-viewer.ts]: technically yes, though browser support is lacking https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach

**Comment by mmalerba** [comment]
changing repo from private --> public borked this pr, starting a new one

**Pull request closed without merging**